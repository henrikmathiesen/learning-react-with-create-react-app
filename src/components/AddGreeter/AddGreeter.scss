@import '../../_settings';

.AddGreeter {
    border: $component-border;
    text-align: center;
    background: $belize-blue;
    color: $almost-black;
    margin: $gutter-md;
    padding: $gutter-md;
}
