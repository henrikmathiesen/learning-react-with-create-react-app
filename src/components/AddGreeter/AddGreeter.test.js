import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import AddGreeter from './AddGreeter';

describe('AddGreeter', () => {

    const mockAddGreeting = jest.fn();
    const component = shallow(
        <AddGreeter addGreeting={mockAddGreeting} />
    );

    // Write a generic snapshot test

    it('Renders and matches our snapshot', () => {
        const component = renderer.create(
            <AddGreeter />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    // Check content

    it('Contains the form', () => {
        expect(component.find('input')).toHaveLength(1);
        expect(component.find('button')).toHaveLength(1);
    });

    // Check interactions

    it('Calls the passed in addGreeting function when add button is clicked', () => {
        component.find('button').simulate('click');
        expect(mockAddGreeting).toBeCalled();
    });

    it('Updates the form when keys are pressed', () => {
        const updateVal = 'foo';
        const event = {
            target: {
                value: updateVal
            }
        }

        component.instance().handleUpdate(event);
        expect(component.state('greetingName')).toEqual(updateVal);
    });

    it('Blanks out the greetingName when the button is clicked', () => {
        const updateVal = 'foo';
        const event = {
            target: {
                value: updateVal
            }
        }

        component.instance().handleUpdate(event);

        expect(component.state('greetingName')).toEqual(updateVal);
        component.find('button').simulate('click');
        expect(component.state('greetingName')).toHaveLength(0);
    });
});
