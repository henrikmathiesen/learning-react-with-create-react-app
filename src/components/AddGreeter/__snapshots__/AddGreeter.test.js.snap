exports[`AddGreeter Renders and matches our snapshot 1`] = `
<div
  className="AddGreeter">
  <input
    onChange={[Function]}
    type="text"
    value="" />
    
  <button
    onClick={[Function]}>
    Add
  </button>
</div>
`;
